android/src/com/mapswithme/maps/search/SearchFragment.java

```diff
@@ -237,15 +222,14 @@ private void hideDownloadSuggest()
   private void updateFrames()
   {
     final boolean hasQuery = mToolbarController.hasQuery();
-    UiUtils.showIf(hasQuery, mResultsFrame);
     Toolbar toolbar = mToolbarController.getToolbar();
     AppBarLayout.LayoutParams lp = (AppBarLayout.LayoutParams) toolbar.getLayoutParams();
     lp.setScrollFlags(hasQuery ? AppBarLayout.LayoutParams.SCROLL_FLAG_ENTER_ALWAYS
                                  | AppBarLayout.LayoutParams.SCROLL_FLAG_SCROLL : 0);
     toolbar.setLayoutParams(lp);
-    if (mFilterController != null)
-      mFilterController.show(hasQuery);
 
+    UiUtils.showIf(hasQuery, mResultsFrame);
```

Comment:
Does it mean that there is no FAB when there are no results? Now users can drag the map and search can start finding objects in other places while dragging.

Reply:
The function `updateFrames()` controls what frame to show (results or categories/history).

I did not change the logic, so when there is no search query, categories/history is shown. When there is a search string, the results are shown instead. At first there is nothing but the list will get populated when search completes. So I simply added a line to show the fab when we show the results frame (empty or not).

Then we have the function `updateResultsPlaceholder()` which is called when the results frame is shown and decides whether to show a placeholder on top (for when search did not find any results). Here I added some logic to show the fab only if search is running and when there are more than 0 search results.

So the fab is shown when user starts typing a query (so no results yet) and stays as long as there is a search in progress or some results displayed. This way we still have the old behavior of clicking "show on map" before search completes and users can pan the map while results appear.

Reply:
Am I correctly understood, that in your current implementation the FAB will disappear when the search finishes with no results? It should not disappear even if there are no results at the search finish.

Reply:
Oh yes I did not know the app could do that. I'll update the PR.